electron-app/main/menu.ts

```diff
@@ -85,4 +86,99 @@ if (process.platform === 'darwin') {
   MENU.push(HELP_MENU);
 }
 
-export default MENU;
+export function popupContextMenu(
+  window: BrowserWindow,
+  isDevelopment: boolean,
+  props: Electron.ContextMenuParams
+) {
+  const editFlags: any = props.editFlags;
+  const hasText = props.selectionText.trim().length > 0;
+  const roleEnabled = (role: string) => editFlags[`can${capitalize(role)}`] && hasText;
```

Comment:
Prefer removing the `any` cast, and doing explicit checks against `editFlags` to maintain typing. It's not that much more code to write, and makes 100% sure it's actually a flag Electron provides.

```
(role: string) => editFlags[`can${capitalize(role)}`] && hasText;
```
can become
```
(flag: keyof Electron.EditFlags) => editFlags[flag] && hasText;
```

Reply:
@wbobeirne i have updated per your comments. Thanks a lot for the code review.